cmake/custom_cxxstandard.cmake

```diff
@@ -6,8 +6,9 @@ if (CMAKE_CXX_COMPILER_ID MATCHES GNU)
     endif()
 
 elseif (CMAKE_CXX_COMPILER_ID MATCHES Intel)
-    if (CMAKE_CXX_COMPILER_VERSION VERSION_LESS "16.0.2")
-        message(FATAL_ERROR "ICPC version must be at least 2016.0.2!")
+    if (CMAKE_CXX_COMPILER_VERSION VERSION_LESS "17.0.0")
+        message(FATAL_ERROR "ICPC version must be at least 2017.0.0 to work with pybind11 2.2.2!")  # v1.2
```

Comment:
I *think* we also determined that 2015 is ok.  Not suggesting we change the minimum, but just making a note in case we encounter user pushback.

Reply:
So there was an earlier episode where we (or pb11) found that 2015 would work, with a hack, and 2016 would work out-of-the-box. That was c. pb11 2.0.0.

And later there was [this](https://github.com/pybind/pybind11/issues/1121) where we found that with pb11 2.1.0 and above that not even Ben could tweak the pb11 code to please Intel 2016. Hence the minimum 2017 in the snippet above.

Which episode are you remembering the 2015-is-ok from?

Reply:
The most unreliable source - my memory.  I think I did some experimentation and got it working but, as you noted, that was likely an earlier PB.  If we get requests for older compiler support, I can mess around with icpc 15 for Linux but not OSX.  Either way, I fully support the icpc17 minimum.